### PR TITLE
Enhance memory stats

### PR DESCRIPTION
**Pull request by ianull**
Add more stats w.r.t memory footprint:
* native heaps
* dalvik + native heaps

Native heap stats can tell the upper bound of db caches, etc.
dalvik + native can represent total run-time memory usage excluding most of mmap files, bss, etc.

### CODE REVIEW

File: javatests/arcs/android/systemhealth/testapp/MemoryStats.kt [hunk 1]
```diff
@@ -12,16 +12,34 @@
...
     val appJvmHeapKbytes: Long
         get() = with(Debug.MemoryInfo().apply { Debug.getMemoryInfo(this) }) {
             dalvikPrivateDirty.toLong()
         }
+
+    /**
+     * App-specific native heap usage.
+     */
+    val appNativeHeapKbytes: Long
+        get() = arcs.core.util.performance.MemoryStats.stat(MemoryIdentifier.NATIVE_HEAP)[0]
```
Comment (cromwellian) [inline javatests/arcs/android/systemhealth/testapp/MemoryStats.kt:36]: optional: you could probably replace this with a static import or an aliased import to void the collision.

File: javatests/arcs/android/systemhealth/testapp/StorageCore.kt [hunk 2]
```diff
@@ -199,7 +199,14 @@ class StorageCore(val context: Context, val lifecycle: Lifecycle) {
...
                         taskManagerEvents.queue.add(
-                            TaskEvent(TaskEventId.MEMORY_STATS, 0, MemoryStats.appJvmHeapKbytes)
+                            TaskEvent(
+                                TaskEventId.MEMORY_STATS,
+                                0,
+                                listOf(
+                                    MemoryStats.appJvmHeapKbytes,
+                                    MemoryStats.appNativeHeapKbytes,
+                                    MemoryStats.allHeapsKbytes
+                                ))
```
Comment (cromwellian) [inline javatests/arcs/android/systemhealth/testapp/StorageCore.kt:209]: I think the closing parens of TaskEvent is supposed to be on a line by itself in the same column as the 'T' of TaskEvent

File: javatests/arcs/android/systemhealth/testapp/MemoryStats.kt (hunk 1 above)
Comment (ianull) [reply javatests/arcs/android/systemhealth/testapp/MemoryStats.kt]: Done and nice suggestion! I totally forgot alias import ;-)

File: javatests/arcs/android/systemhealth/testapp/StorageCore.kt (hunk 2 above)
Comment (ianull) [reply javatests/arcs/android/systemhealth/testapp/StorageCore.kt]: Done.

**Pull request merged**